### PR TITLE
docs: adding a terraform tutorial to the docs

### PR DESCRIPTION
**Pull request by AnaisUrlichs**
We received a lot of questions by the tfsec community on how to move over to Trivy, how different Trivy for terraform scanning is and similar.

### CODE REVIEW

**Comment by AnaisUrlichs** [comment]
@giorod3 could you please take a look and let me know anything else that you would like to have included?

File: docs/tutorials/misconfiguration/terraform.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+This tutorial is focused on ways Trivy can scan Terraform IaC configuration files. 
+
+**A note to tfsec users** 
+We have been consolidating all of our scanning-related efforts in one place, and that is Trivy. Over the past year, tfsec has laid the foundations to Trivy's IaC & misconfigurations scanning capabilities, including Terraform scanning, which has been natively supported in Trivy for a long time now. 
+Going forward we want to encourage the tfsec community to transition over to Trivy. Moving to Trivy gives you the same excellent Terraform scanning engine, with some extra benefits: 
+
+- Access to more languages and features in the same tool. 
+- Access to more integrations with tools and services through the rich ecosystem around Trivy. 
+- Commercially supported by Aqua as well as by a the passionate Trivy community.
```
Comment (simar7) [inline docs/tutorials/misconfiguration/terraform.md:11]:
```suggestion
- Commercially supported by Aqua as well as by the passionate Trivy community. 
```

File: docs/tutorials/misconfiguration/terraform.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+``` 
+trivy conf --tf-vars dev.terraform.tfvars ./terraform-infra 
+``` 
+### Custom Policy 
+
+We have lots of example in the [documentation](https://aquasecurity.github.io/trivy/latest/docs/misconfiguration/custom/) on how you can write and pass custom Rego policies into terraform misconfiguration scans. 
+
+## Secret scans and vulnerability scans as part of misconfiguration scans 
+
+The trivy config` command does not perform secrete and vulnerability checks out of the box. However, you can specify as part of your `trivy fs` scan that you would like to scan you terraform files for exposed secrets through an additional flag:
```
Comment (simar7) [inline docs/tutorials/misconfiguration/terraform.md:102]:
```suggestion
The `trivy config` command does not perform secrete and vulnerability checks out of the box. However, you can specify as part of your `trivy fs` scan that you would like to scan you terraform files for exposed secrets through an additional flag: 
```

**Comment by simar7** [comment]
hi @AnaisUrlichs any update on this?

**Comment by AnaisUrlichs** [comment]
@simar7 could you please have another look?

**Review by simar7 — APPROVED** [review]
lgtm

**Review by simar7 — APPROVED** [review]

File: docs/tutorials/misconfiguration/terraform.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+trivy config <any flags you want to use> <file or directory that you would like to scan> 
+``` 
+
+The `trivy config` command can scan Terraform configuration, CloudFormation, Dockerfile, Kubernetes manifests, and Helm Charts for misconfiguration. Trivy will compare the configuration found in the file with a set of best practices.  
+
+- If the configuration is following best practices, the check will pass,  
+- If the configuration does not define some configuration according to best practices, the default is used, 
+- If the configuration does not follow best practices, the check will fail.  
+
+## Using the `trivy config` command
```
Comment (itaysk) [inline docs/tutorials/misconfiguration/terraform.md:25]: seems like redundant title with previous title

File: docs/tutorials/misconfiguration/terraform.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+Command structure: 
+``` 
+trivy config <any flags you want to use> <file or directory that you would like to scan> 
+``` 
+
+The `trivy config` command can scan Terraform configuration, CloudFormation, Dockerfile, Kubernetes manifests, and Helm Charts for misconfiguration. Trivy will compare the configuration found in the file with a set of best practices.  
+
+- If the configuration is following best practices, the check will pass,  
+- If the configuration does not define some configuration according to best practices, the default is used,
```
[the remark this answers is not in the document]
Comment (AnaisUrlichs) [reply docs/tutorials/misconfiguration/terraform.md:22]: Basically, if you e.g. do not set the `PullPolicy` Trivy will use the default. I rewrote it:
> If the configuration does not define the resource of some configuration, Trivy will assume the default configuration for the resource creation is used. In this case, the check might fail.

File: docs/tutorials/misconfiguration/terraform.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+``` 
+
+### Passing tf.vars files into `trivy config` scans 
+
+You can pass tf-vars files to Trivy to override default values found in the Terraform HCL code. More information are provided [in the documentation.](https://aquasecurity.github.io/trivy/latest/docs/misconfiguration/options/values/) 
+
+``` 
+trivy conf --tf-vars terraform.tfvars ./
+``` 
+### Custom Policy
```
[the remark this answers is not in the document]
Comment (AnaisUrlichs) [reply docs/tutorials/misconfiguration/terraform.md:95]: we still call them policy in the docs https://aquasecurity.github.io/trivy/v0.42/docs/scanner/misconfiguration/custom/

**Review by itaysk — APPROVED** [review]

**Pull request merged**